### PR TITLE
PCHR-2613: Customize quicksearch field

### PR DESCRIPTION
**Pull request by AkA84**
## Overview
This PR introduces customizations to the quicksearch field in the top right corner of a CiviHR site

## Before
before-menu

The field was just a static field, with a "Contacts" placeholder

## After
after-menu

The field is now completely collapsed until the user hovers overs it, and stays expanded if there is an ongoing search. The placeholder text is changed to "Quick Search"

### CODE REVIEW

File: hrui/js/src/hrui.js [hunk 1]
```diff
@@ -171,4 +173,25 @@
...
+   * The class is removed only when the element
+   * loses the hover AND it is empty (= there is no ongoing search)
+   */
+  function addCustomClassToQuickSearchField () {
+    $('#crm-qsearch').hover(
+      function () {
+        $(this).addClass('search-ongoing');
+      },
+      function () {
+        if (!$('#sort_name_navigation').val().length &&
```
Comment (deb1990) [inline hrui/js/src/hrui.js]: Creating some variables here would make the code more readable.
```js
var onGoingSearchResultItems = $('.crm-quickSearchField:visible', '#root-menu-div').length;
// same goes for $('#sort_name_navigation').val().length

File: hrui/js/src/hrui.js [hunk 2]
```diff
@@ -171,4 +173,25 @@
...
   });
+
+  /**
+   * Adds a custom class to the quicksearch field
+   * so that custom behaviour can be applied to it
+   *
+   * The class is removed only when the element
+   * loses the hover AND it is empty (= there is no ongoing search)
+   */
+  function addCustomClassToQuickSearchField () {
```
Comment (deb1990) [inline hrui/js/src/hrui.js]: can we name it `toggleHoverClassofQuickSearchField`?

File: hrui/js/src/hrui.js [hunk 3]
```diff
@@ -91,6 +91,21 @@
   }
 
   /**
+   * Changes the placeholder text of the quicksearch field
+   */
+  function changeQuickSearchFieldPlaceholder () {
+    $('#crm-qsearch .ui-autocomplete-input').attr('placeholder', 'Quick Search');
```
Comment (deb1990) [inline hrui/js/src/hrui.js]: Can we do it from the markup?

File: hrui/js/src/hrui.js [hunk 4]
```diff
@@ -1,11 +1,115 @@
+/* global Ps */
+
 // Copyright CiviCRM LLC 2013. See http://civicrm.org/licensing
 (function ($, _) {
+  $('.CRM_HRRecruitment_Form_Application').addClass('crm-form-block');
```
Comment (deb1990) [inline hrui/js/src/hrui.js]: This file is a mess right now, with no proper structure to the code. Can we clean this up a bit.
```js
(function init () {
  $(document).on('crmLoad', crmOnLoad);
  $(document).ready(onReady);
})();

//Keep all the private functions below
// Also can we break crmOnLoad into smaller chunks

File: hrui/js/src/hrui.js (hunk 3 above)
Comment (AkA84) [reply hrui/js/src/hrui.js]: no, the markup is core markup, we can't change it

**Comment by igorpavlov** [comment]
Has this solution been tested on mobile devices with touch events? If you need help testing on iOS please let me know.

**Comment by igorpavlov** [comment]
The code looks fine.

**Comment by igorpavlov** [comment]
It works like this on touch devices:
1) you tap - it expands
2) you tap again - it focuses
3) you enter the text with a virtual keyboard, hit virtual enter and it searches
4) when you tap anywhere but on the search object, it collapses

cc @AkA84

**Pull request merged**